addons/koji/common/src/main/java/org/commonjava/indy/koji/content/KojiMavenMetadataProvider.java

```diff
@@ -194,7 +194,16 @@ public Metadata getMetadata( StoreKey targetKey, String path )
                                 logger.debug( "Skipping non-POM: {}", archive.getFilename() );
                                 continue;
                             }
-                            SingleVersion singleVersion = VersionUtils.createSingleVersion( archive.getVersion() );
+                            SingleVersion singleVersion = null;
+                            try
+                            {
+                                singleVersion = VersionUtils.createSingleVersion( archive.getVersion() );
+                            }
+                            catch ( InvalidVersionSpecificationException ivse )
+                            {
+                                logger.info( "Skipping mal-formatted version: {}, archive: {}", archive.getVersion(), archive );
```

Comment:
Just a minor thing here - the KojiArchiveInfo.toString() prints all field values, so printing version first and then everything does not make much sense. I would probably print complete version along with relpath and buildId. Also using warn level seems more sensible in this case.

Reply:
make sense. I will update it.